improved language switching, fixed transparent navigation bar bug

Iferencak opened the pull request.

mosadialiou commented on loafwallet/UnstoppableDomainView.swift:
```diff
@@ -73,7 +76,15 @@ struct UnstoppableDomainView: View {
+                         // Description: the minmum domain length is 4 e.g.; 'a.zil'
+                         // Enabling the button until the domain string is at least 4 chars long
+                        
+                         shouldDisableLookupButton = currentString.count > 4 ? false : true
```
Just write 
`shouldDisableLookupButton = currentString.count < 4`

mosadialiou reviewed (changes requested):
BTW translate the "Lookup" text here 
https://github.com/litecoin-foundation/loafwallet-ios/blob/923fb913a77c09ea5415ebf0eb1428cc0aacd67d/loafwallet/UnstoppableDomainView.swift#L68

mosadialiou reviewed (approved):
LGTM

Iferencak committed:
Improved language selection

kcw-grunt commented on loafwallet.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
```
Thought  I .gitignored xcshareddata? @Iferencak .  Is this a new setting?  Should I migrate with it?  Whats the benefit?

Iferencak replied:
Honestly I'm not sure how this got tracked.. I didn't touch .gitignore 🤷

kcw-grunt commented on loafwallet/AppDelegate.swift:
```diff
@@ -10,6 +10,16 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
 	func application(_ application: UIApplication, didFinishLaunchingWithOptions _: [UIApplicationLaunchOptionsKey: Any]?) -> Bool
 	{
+		UINavigationBar.appearance().tintColor = .darkText
```
TYVM

kcw-grunt commented on loafwalletTests/Language Selection Tests/LanguageSelectionTests.swift:
```diff
@@ -0,0 +1,30 @@
+		let initialLanguage = UserDefaults.selectedLanguage
+		XCTAssertEqual(initialLanguage, "en")
+
+		let spannish = "es"
```
Microsopic nit  🔬 spannish should be spanish

Iferencak replied:
Oops... On it

kcw-grunt commented on loafwallet/BundleExtension.swift:
```diff
@@ -0,0 +1,46 @@
+import Foundation
+
+var _bundle: UInt8 = 0
+
```
I love this solution.  If I understand correctly, BundleEx does the heavy lifting of converting all the strings to the chosen language in a DQ?  Its quite smooth

Only thing is I generally prefer `guard let ` rather than forced unwrap. 

Can we do :

```
guard let bundle = Bundle(path: Bundle.main.path(forResource: temp, ofType: "lproj") else {
	 LWAnalytics.logEvent(itemName: ._20200112_ERR)
return
}
```

Maybe it will never be called but it will help us sleep a little better.

Iferencak replied:
Sure thing, looks like i missed it after I finished prototyping. On it.

kcw-grunt commented on loafwallet/LanguageSelectionViewController.swift:
```diff
@@ -0,0 +1,57 @@
+class LanguageSelectionViewController: UITableViewController {
+	let viewModel = LanguageSelectionViewModel()
+
+	var titleLabel: UILabel!
```
🥲
Not really into force unwrap....it's hidden poison. 

Why not just :

```
var titleLabel = UILabel()
```

Iferencak replied:
👍

Iferencak replied:
I agree completely, I hate force unwraping, sry. 😱

kcw-grunt commented on loafwallet/NonUSTabBarViewController.swift:
```diff
@@ -71,12 +68,33 @@ class NonUSTabBarViewController: UIViewController, Subscriber, Trackable, UITabB
 	}
 
+	@objc
+	func languageChanged() {
```
smooth operator
🕺

kcw-grunt reviewed (changes requested):
This so smooth.  Just saw some tiny issues in the code that I noted and saw a bug in the switch.

When the user switches languages and you view the History, the sublabel is not switching and stays the old language:

|Note stale 'Complete'|
|----------------------|
|![history-view-complete-bug](https://user-images.githubusercontent.com/2899463/212498611-39d30755-32a2-4737-ab0a-1bee1dc01684.gif)|

Iferencak committed:
Fixed transaction completion localization, removed force unwrapings, fixing typos

kcw-grunt committed:
autoformat

- BartyCrouch has a bug where the NSLocalizaedString makes extractLocStrings fail.
The workaround is detailed here:
https://github.com/FlineDev/BartyCrouch#exclude-specific-views--nslocalizedstrings-from-localization

kcw-grunt committed:
resolved flaky compile issue

- BC and extractLocStrings did not get along
Barty Crouch have an open issue as extractLocStrings is deprecated by Apple so the script is barfing.
The notes on the issue are here: https://github.com/FlineDev/BartyCrouch/issues/135#issuecomment-772379904

The other issue was the NSNotificationCenter was not happy and the work around was:
- deinit and remove observers for the new post
- remove the assertFailure (which I added) and replaced it with an LWAnalytics error event
- added a Error NSlog message
- also moved all the NSN to its own file NSNExtension

kcw-grunt committed:
bump version

- polished language view
